src/test/java/org/logstash/netty/SslContextBuilderTest.java

```diff
@@ -37,4 +89,141 @@ public void getDefaultCiphers() throws Exception
         List<String> availableCiphers = Arrays.asList(ssf.getSupportedCipherSuites());
         assertThat(Arrays.asList(defaultCiphers), Every.everyItem(isIn(availableCiphers)));
     }
+
+    @Test
+    public void testSetClientAuthentication() {
+        final SslContextBuilder sslContextBuilder = createSslContextBuilder();
+        final String[] certificateAuthorities = {CA};
+
+        sslContextBuilder.setClientAuthentication(SslClientVerifyMode.REQUIRED, certificateAuthorities);
+
+        assertThat(sslContextBuilder.getVerifyMode(), is(SslClientVerifyMode.REQUIRED));
+        assertThat(Arrays.asList(sslContextBuilder.getCertificateAuthorities()), Every.everyItem(isIn(certificateAuthorities)));
+    }
+
+    @Test
+    public void testSetClientAuthenticationWithRequiredAndNoCertAuthorities() {
+        assertSetClientAuthenticationThrowWithNoCerts(SslClientVerifyMode.REQUIRED);
+    }
+
+    @Test
+    public void testSetClientAuthenticationWithOptionalAndNoCertAuthorities() {
+        assertSetClientAuthenticationThrowWithNoCerts(SslClientVerifyMode.OPTIONAL);
+    }
+
+    @Test
+    public void testSetClientAuthenticationWithNone() {
+        final SslContextBuilder sslContextBuilder = createSslContextBuilder();
+
+        sslContextBuilder.setClientAuthentication(SslClientVerifyMode.NONE, new String[0]);
+        assertThat(sslContextBuilder.getVerifyMode(), is(SslClientVerifyMode.NONE));
+        assertThat(sslContextBuilder.getCertificateAuthorities(), arrayWithSize(0));
+
+        sslContextBuilder.setClientAuthentication(SslClientVerifyMode.NONE, null);
+        assertThat(sslContextBuilder.getVerifyMode(), is(SslClientVerifyMode.NONE));
+        assertThat(sslContextBuilder.getCertificateAuthorities(), nullValue());
+    }
+
+    @Test
+    public void testDefaultVerifyMode() {
+        final SslContextBuilder sslContextBuilder = createSslContextBuilder();
+        assertThat(sslContextBuilder.getVerifyMode(), is(SslClientVerifyMode.NONE));
+    }
+
+    private void assertSetClientAuthenticationThrowWithNoCerts(SslClientVerifyMode mode) {
+        final SslContextBuilder sslContextBuilder = createSslContextBuilder();
+        final String expectedMessage = "Certificate authorities are required to enable client authentication";
+
+        try {
+            sslContextBuilder.setClientAuthentication(mode, new String[0]);
+            Assert.fail("No exception was thrown");
```

Comment:
All the other `Assert` methods are statically imported, so maybe we could use the same pattern and move to `fail()`